File: leetcode/392_issubsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if s is None or len(s) == 0:
            return True
            
        indices = [-1 for i in range(len(s) + 1)]
        
        for i in range(len(s)):
            if i > 0 and indices[i] == -1:
                return False
            for j in xrange(indices[i] + 1, len(t)):
                if t[j] == s[i]:
                    indices[i + 1] = j
                    break
        return True if indices[len(s)] >= 0 else False

File: leetcode/392_issubsequence.ts
function isSubsequence(s: string, t: string): boolean {
  let sP = 0
  let tP = 0

  while (sP < s.length && tP < t.length) {
    if (s[sP] === t[tP]) {
      sP += 1
    }
    tP += 1
  }

  if (sP === s.length) return true
    
  return false
};
